aten/src/ATen/native/cpu/ReduceOpsKernel.cpp

```diff
@@ -133,6 +133,16 @@ static void sum_kernel_impl(TensorIterator& iter) {
       });
 }
 
+static void nansum_kernel_impl(TensorIterator& iter) {
+  if (iter.dtype() == ScalarType::Half) {
+    binary_kernel_reduce(iter, NanSumOps<float>{}, float{0});
+  } else {
+    AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "nansum_cpu", [&] {
```

Comment:
Restricting this to floating types seems OK but can it not handle complex dtypes as written?

Reply:
I guess it would but I was thinking of doing it in a different PR with relevant tests.